internal/auth/authenticator_oidc_token_expressions.go

```diff
@@ -0,0 +1,66 @@
+package auth
+
+import (
+	"errors"
+
+	"github.com/tidwall/gjson"
+)
+
+// Lowercase is intentional to make it easier to write in HAProxy configuration
+const (
+	operationIn           string = "in"
+	operationNotIn        string = "notin"
+	operationExists       string = "exists"
+	operationDoesNotExist string = "doesnotexist"
+)
+
+var ErrParseTokenExpressionRequest = errors.New("failed to parse OAuthTokenExpression")
+
+type OAuthTokenExpressionOperation int
+
+func (te OAuthTokenExpressionOperation) String() string {
+	switch te {
+	case in:
+		return operationIn
+	case notIn:
+		return operationNotIn
+	case exists:
+		return operationExists
+	case doesNotExist:
+		return operationDoesNotExist
+	}
+
+	return ""
+}
+
+const (
```

Comment:
To simplify, you could probably merge this with the constants

```const (
	operationIn           string = "in"
	operationNotIn        string = "notin"
	operationExists       string = "exists"
	operationDoesNotExist string = "doesnotexist"
)```

so that during the assignement it becomes as simple as

```
			expr.Operation = val[0].value
```

Reply:
I am sorry, I might have misunderstood your idea but I can not see how I can simplify this.

May I ask you to explain me?